bowtie/_cli.py

```diff
@@ -1684,6 +1708,623 @@ async def info(
+                        f"directory of {input.name} as per listed by its "
+                        "matrix-versions.json file.",
+                    )
+                    ctx.exit(EX.DATAERR)
+
+                progress = Progress(
+                    TextColumn("[bold blue]{task.description}"),
+                    SpinnerColumn(finished_text=""),
+                    BarColumn(bar_width=None),
+                    TextColumn(
+                        "[progress.percentage]{task.percentage:>3.0f}%",
+                    ),
+                    "•",
+                    MofNCompleteColumn(),
+                    "•",
+                    TimeElapsedColumn(),
+                    console=console.Console(),
+                    transient=True,
+                )
+                dialects = dialects or Dialect.known()
+                pretty_names_str = pretty_names_str_for(dialects)
+                total_files = len(versions) * len(list(dialects))
+                task = progress.add_task(
+                    description=(
+                        f"Preparing to parse all versioned reports of "
+                        f"{id} for {pretty_names_str} found in {input.name}"
+                    ),
+                    total=total_files,
+                )
+
+                actual_parsed_files = [0]
+                versioned_reports: Iterable[
+                    tuple[str, Dialect, _report.Report]
+                ] = []
+
+                with progress:
+                    for version in versions:
+                        for dialect in dialects:
+                            progress.update(
+                                task,
+                                description=(
+                                    f"Parsing: "
+                                    f"v{version}/{dialect.short_name}.json"
+                                ),
+                            )
+                            try:
+                                report_content = tar.extractfile(
+                                    f"./{id}/v{version}/{dialect.short_name}.json",
+                                )
+                            except KeyError:
+                                versioned_reports.append(
+                                    (
+                                        version,
+                                        dialect,
+                                        _report.Report.empty(dialect=dialect),
+                                    ),
+                                )
+                            else:
+                                if report_content:
+                                    versioned_reports.append(
+                                        (
+                                            version,
+                                            dialect,
+                                            _report.Report.from_serialized(
+                                                cast(IO[Any], report_content),
```

Comment:
This is wrong. You are passing an iterable of bytes to something that claims to take an iterable of str. You calling `cast` just throws away information to make it seem like this type is correct, when it isn't. If you want to "lie" to the type checker, just use ignore. But here, that's not what you should do.

You should have instead either:

* Change the type signature of `from_serialized` to say it accepts an iterable of either, because really it calls json.loads, which does take both.
* Or instead of passing the bytes stream, use `io.TextIOWrapper` to get an iterable of str file-like object instead.